### ISSUE TITLE
Polyfill of promises does not work.

### ISSUE THREAD

**Issue by antoncl** [less/less.js#2322]
It would appear that that the 2.x version of less relies on Promises, and from glancing at the code, I can see that an attempt to implement a polyfill of promises has been made, probably in order to be compatible with platforms where promises aren't available yet.

Unfortunately that implementation seems defective. Specifically in line 680 a resolve is called without any parameters. This invocation causes the polyfill (in lines, 9126-9140) to ignore the invocation of 'then'; specifically the 'then' part of the PageLoadFinished handler is never invoked.

In short, the browser based functionality will only work on browsers which has inbuilt support for promises.
